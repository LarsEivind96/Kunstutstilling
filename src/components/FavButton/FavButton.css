.FavButton {
  flex: 1;
  background-color: transparent;
  color: white;
  border: none;
  text-align: center;
  text-decoration: none;
  font-size: 6vw;
  transition-duration: 0.4s;
  cursor: pointer;
  outline: none;
  margin-top: 0.5vw;
}

.FavButton:hover {
  opacity: 0.4;
}

@media screen and (max-width: 550px) {
  .FavButton {
    font-size: 10vw;
  }
}
