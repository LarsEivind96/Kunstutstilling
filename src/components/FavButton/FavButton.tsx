import React from "react";

interface ButtonProps {
  index: string;
  label: string;
}

interface ButtonState {}

class FavButton extends React.Component<ButtonProps, ButtonState> {
  // Save favorite
  saveFavorite = () => {
    localStorage.setItem("favorite", this.props.index);
  };

  render() {
    return (
      <button data-testid="favbut" className="FavButton" onClick={this.saveFavorite}>
        {this.props.label}
      </button>
    );
  }
}

export default FavButton;
