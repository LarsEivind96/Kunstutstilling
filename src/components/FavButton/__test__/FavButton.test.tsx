import React from "react";
import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import FavButton from "../FavButton";

// Clean up after each test
afterEach(cleanup);

// Test if the button renders on a div without failure
it("button renders without failure", () => {
  const div = document.createElement("div");
  ReactDOM.render(<FavButton index="2" label="Test label" />, div);
});

// Snapshot test 1
it("correlates to snapshot 1", () => {
  const { container } = render(<FavButton index="1" label="Save as favorite" />);
  expect(container).toMatchSnapshot();
});

// Snapshot test 2
it("correlates to snapshot 2", () => {
  const { container } = render(<FavButton index="3" label="Next SVG" />);
  expect(container).toMatchSnapshot();
});
