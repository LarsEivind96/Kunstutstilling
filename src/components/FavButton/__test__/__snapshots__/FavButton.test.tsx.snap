// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correlates to snapshot 1 1`] = `
<div>
  <button
    class="FavButton"
    data-testid="favbut"
  >
    Save as favorite
  </button>
</div>
`;

exports[`correlates to snapshot 2 1`] = `
<div>
  <button
    class="FavButton"
    data-testid="favbut"
  >
    Next SVG
  </button>
</div>
`;
